stock_release_channel_partner_delivery_window/models/stock_release_channel.py

```diff
@@ -16,27 +22,83 @@ class StockReleaseChannel(models.Model):
+                    days = 2
+                delivery_date_tz += timedelta(days=days)
+                delivery_date = self._naive(delivery_date_tz, reset_time=days)
+                delivery_date = yield delivery_date
+
+        while True:
+            delivery_date_tz = self._localize(delivery_date, tz=tz)
+            weekday = delivery_date_tz.weekday()
+            for inc in range(8):
+                windows = partner.delivery_time_window_ids.filtered(
+                    lambda w: str(weekday + inc)
+                    in w.time_window_weekday_ids.mapped("name")
+                    and (inc or w.get_time_window_end_time() >= delivery_date_tz.time())
+                )
+                if windows:
+                    w = windows[0]
+                    break
+            else:
+                # There is no time window, we consider any date valid
+                while True:
+                    delivery_date = yield delivery_date
+            delivery_date_tz = datetime.combine(
+                (delivery_date_tz + timedelta(days=inc)).date(),
+                max(w.get_time_window_start_time(), delivery_date_tz.time())
+                if not inc
+                else w.get_time_window_start_time(),
+                tzinfo=delivery_date_tz.tzinfo,
+            )
+            delivery_date = self._naive(delivery_date_tz)
+            delivery_date = yield delivery_date
+
     def filter_release_channel_partner_window(self, picking, partner):
-        channels = self
-        if (
-            not partner.delivery_time_preference
-            or partner.delivery_time_preference == "anytime"
-        ):
-            return channels
-
-        for channel in self:
-            if not channel.shipment_date:
-                continue
-            shipment_datetime = fields.Datetime.to_datetime(channel.shipment_date)
-            if channel.process_end_date:
-                shipment_datetime = shipment_datetime.replace(
-                    hour=channel.process_end_date.hour,
-                    minute=channel.process_end_date.minute,
+        if picking.sale_id.commitment_date:
+            # Date is forced on SO, don't filter
+            return self
+        if partner.delivery_time_preference == "anytime":
+            return self
+        if partner.delivery_time_preference == "workdays":
+            open_days = set(range(5))
+        else:
+            open_days = set(
+                map(
+                    int,
+                    partner.delivery_time_window_ids.time_window_weekday_ids.mapped(
+                        "name"
+                    ),
                 )
-            if (
-                channel.respect_partner_delivery_time_windows
-                and not picking.sale_id.commitment_date
-                and not partner.is_in_delivery_window(shipment_datetime)
-            ):
-                channels -= channel
-        return channels
+            )
```

Comment:
please use the method you defined here 
https://github.com/OCA/stock-logistics-workflow/pull/1940
```suggestion
open_days = partner.delivery_time_weekdays
```

Reply:
that was already solved by https://github.com/OCA/wms/pull/1025, i had to rebase